src/OpenLoco/Map/BuildingTile.cpp

```diff
@@ -1,24 +1,55 @@
+#include "../CompanyManager.h"
 #include "../Game.h"
 #include "../GameCommands/GameCommands.h"
 #include "../Objects/BuildingObject.h"
 #include "../StationManager.h"
 #include "../TownManager.h"
+#include "../Ui/WindowManager.h"
 #include "../ViewportManager.h"
 #include "AnimationManager.h"
 #include "Tile.h"
 #include "TileManager.h"
 
 namespace OpenLoco::Map
 {
+    static loco_global<uint32_t, 0x01135C38> dword_1135C38;
+
+    // 0x00497DC1
     static void sub_497DC1(const Map::Pos2& loc, uint32_t population, uint32_t unk1, uint16_t rating, uint16_t unk3)
     {
-        registers regs;
-        regs.edi = population;
-        regs.esi = unk1;
-        regs.ebp = rating | (unk3 << 16);
-        regs.ax = loc.x;
-        regs.cx = loc.y;
-        call(0x00497DC1, regs);
+        auto res = TownManager::getClosestTownAndUnk(loc);
+        if (res == std::nullopt)
+        {
+            dword_1135C38 = 0xFFFFFFFF;
+            return;
+        }
+        auto townId = res->first;
+        auto town = TownManager::get(townId);
+        dword_1135C38 = static_cast<uint32_t>(town->id());
```

Comment:
this is storing the actual town ptr not the id. Just change the type of `dword_1135C38` to `Town*`